rs/ledger_suite/tests/sm-tests/src/lib.rs

```diff
@@ -2668,6 +2668,14 @@ pub fn test_upgrade_serialization(
                         Ok(_) => {
                             if !downgrade_to_mainnet_should_succeed {
                                 panic!("Downgrade from memory manager directly to mainnet should fail (since mainnet is V0)!")
+                            } else {
+                                // In case this succeeded, we need to upgrade the ledger back to
+                                // the next version, so that the subsequent upgrade is from
+                                // `ledger_wasm_nextmigrationversionmemorymanager` to
+                                // `ledger_wasm_current`, rather than from `ledger_wasm_mainnet` to
+                                // `ledger_wasm_current` (currently, from V2 -> V1, rather than from
+                                // V1 -> V1).
+                                test_upgrade(ledger_wasm_nextmigrationversionmemorymanager);
```

Comment:
Here we're jumping V0 -> V2 which is a transition we might not need to support: how about going V0 -> V1 -> V2?

Reply:
Good catch, thanks - done!